### ISSUE TITLE
building with ant fails in get-buck-info

### ISSUE THREAD

**Issue by matanster** [facebook/buck#1812]
Over a fresh git pull of buck. Any advice?

```
gen-buck-info:
     [exec] Traceback (most recent call last):
     [exec]   File "programs/gen_buck_info.py", line 51, in <module>
     [exec]     sys.exit(main(sys.argv))
     [exec]   File "programs/gen_buck_info.py", line 48, in main
     [exec]     indent=2)
     [exec]   File "...../anaconda3/lib/python3.6/json/__init__.py", line 179, in dump
     [exec]     for chunk in iterable:
     [exec]   File "...../anaconda3/lib/python3.6/json/encoder.py", line 430, in _iterencode
     [exec]     yield from _iterencode_dict(o, _current_indent_level)
     [exec]   File "...../anaconda3/lib/python3.6/json/encoder.py", line 404, in _iterencode_dict
     [exec]     yield from chunks
     [exec]   File "...../anaconda3/lib/python3.6/json/encoder.py", line 437, in _iterencode
     [exec]     o = _default(o)
     [exec]   File "...../anaconda3/lib/python3.6/json/encoder.py", line 180, in default
     [exec]     o.__class__.__name__)
     [exec] TypeError: Object of type 'bytes' is not JSON serializable
```

Any quick advice?

**Comment by styurin** [comment]
Try Python 2.7?

**Comment by matanster** [comment]
Right. This is implied in the prerequisites section. Unfortunately I am not sure how to tell ant to use python 2.7 when python 3 is the default on the machine.

**Comment by ttsugriy** [comment]
@styurin, given that Python 2.7 is going away soon, I don't think it's unreasonable to fix Buck's code to make it forward compatible with Python 3. If nothing else, we should encourage users to use it, since it brings speed and usability improvements.

**Comment by ttsugriy** [comment]
the problem is that `gen_buck_info.py` is using different ways to compute Buck version some of which return `str` while others return `bytes`. It's unfortunate that we're using Python, but we'll definitely have to fix this in Buck.

**Comment by matanster** [comment]
I wonder whether it stumbles on this only for me....

**Comment by matanster** [comment]
I guess I should. Actually I just wish to have it for Nuclide
<https://nuclide.io/docs/languages/cpp/>, whereas the jitpack page of it
<https://jitpack.io/#facebook/buck/v2018.02.16.01> only (?) implies a
development project use case installation.
Any idea how would Nuclide use it? maybe I should ask that on the Nuclide
project....

**Comment by ttsugriy** [comment]
@matanster, there is no such thing as Buck for Nuclide - there is just Buck. Nuclide can use it underneath for some of its plugins, but it's the same binary and the same set of steps to install.

As to why Facebook and all other companies with huge internal repositories use Buck-like build tools instead of Make/Maven/Gradle-like ones, you can read on our main page https://buckbuild.com/ In short - those tools are slow and not scalable, but Buck is fast and scales to huge projects that are built from source.

**Comment by matanster** [comment]
Well I'm lost on how to install buck for Nuclide's C++ support. Its mentioned it is a prerequisite, but the buck download page only mentions how to include it in development projects. I'll fetch my answer on the nuclide github repo ― thanks for all your support so far!

**Comment by ttsugriy** [comment]
@matanster, as buckbuild.com says the easiest way to install Buck is to run `brew install buck`.

**Comment by matanster** [comment]
Yep, on mac. Less so linux, but thanks anyway.

**Comment by ttsugriy** [comment]
Our release page has deb package for Linux in case it works for you and
jitpack works for all OSes and distributions.

**Comment by matanster** [comment]
I wasn't aware of the deb file on the releases page (
https://github.com/facebook/buck/releases). it works!
As to jitpack I've no idea how to install from it, the jitpack page only
mentions how to include buck as a project dependency.... perhaps you might
comment

**Comment by styurin** [comment]
Fixed in 4ba983ea268d37a301123b6b8ad6e20f843d1aef
